modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -185,6 +185,21 @@ public ObjectField addOrUpdateCustomObjectField(
 				objectFieldSettings);
 		}
 
+		if (Objects.equals(businessType, "Relationship")) {
+			ObjectDefinition objectDefinition =
+				_objectDefinitionPersistence.findByPrimaryKey(
+					objectDefinitionId);
+
+			if (objectDefinition.isRootDescendantNode() &&
+				(Objects.equals(readOnly, "true") ||
```

Comment:
@larissacribeiro please use the constant instead: `ObjectFieldConstants.READ_ONLY_TRUE`